features/core/src/main/java/org/apache/karaf/features/internal/service/FeaturesServiceImpl.java

```diff
@@ -234,14 +181,14 @@ public FeaturesServiceImpl(Bundle bundle,
                                String blacklisted,
                                boolean configCfgStore) {
         this.bundle = bundle;
-        this.bundleContext = bundleContext;
+        this.bundleContext = bundle == null ? null : bundle.getBundleContext();
```

Comment:
I'd rather keep the bundleContext as a parameter.
The reason is that bundle.getBundleContext() can possibly throw an exception.